test/types/community/collection/findX.test-d.ts

```diff
@@ -305,6 +305,30 @@ expectType<WithId<{ a: number; b: string }> | null>(
   ).value
 );
 
+// Including { includeResultMetadata: false } option will change the return type
```

Comment:
Since we now have specific overloads for all the `findOneAnd*` methods, I think we should add type tests for each.

Reply:
Added tests for all overloads.